### PR TITLE
Fix for insertid not returning what it's supposed to

### PR DESCRIPTION
**Pull request by chrisdavenport**
Pull Request for Issues #9534, #9572 and #9582.
#### Summary of Changes

Same as #9601 but code has been moved to the MySQL-specific PDO driver.
#### Testing Instructions

You need a site which fails under the circumstances described in the issues above.  The easiest is probably #9572.   Apply the PR then check that the issue is resolved.

### CODE REVIEW

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on e5a9742ecc3dcbc6192e8583383b07e78ec8f946<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9610">issues.joomla.org/joomla-cms/9610</a>.</sub>

File: libraries/joomla/database/driver/pdomysql.php [hunk 1]
```diff
@@ -191,6 +191,20 @@ public function dropTable($tableName, $ifExists = true)
...
+	 *
+	 * @return  string  The value of the auto-increment field from the last inserted row.
+	 *
+	 * @since   12.1
+	 */
+	public function insertid()
+	{
+		$this->connect();
+
+		return $this->setQuery('SELECT LAST_INSERT_ID() AS id')->execute()->fetchObject()->id;
```
Comment (bembelimen) [inline libraries/joomla/database/driver/pdomysql.php]: Two comments:
- Perhaps here you can use JQuery:
`$query = $this->getQuery(true)->select('LAST_INSERT_ID() AS ' . $this->qn('id'));`
- I guess a shorter call would be:
`(int) $this->setQuery($query)->loadResult();`
so:
```php
public function insertid()
{
    $query = $this->getQuery(true)->select('LAST_INSERT_ID() AS ' . $this->qn('id'));
    return (int) $this->setQuery($query)->loadResult();
}

**Comment by wilsonge** [comment]
I found the root cause of this and committed the fix after getting it approved my @mbabker

**Comment by chrisdavenport** [comment]
Well done. :-)

**Comment by Fedik** [comment]
@wilsonge :+1:

**Pull request closed without merging**